src/main/java/ch/njol/skript/conditions/CondIsWhitelisted.java

```diff
@@ -19,72 +19,74 @@
 package ch.njol.skript.conditions;
 
 import org.bukkit.Bukkit;
-import org.bukkit.entity.Player;
+import org.bukkit.OfflinePlayer;
 import org.bukkit.event.Event;
 import org.eclipse.jdt.annotation.Nullable;
 
 import ch.njol.skript.Skript;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Condition;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.Kleenean;
 
+import java.util.ArrayList;
+import java.util.List;
+
 @Name("Is Whitelisted")
-@Description("Whether or not the server or a player is whitelisted.")
-@Examples({"if server is whitelisted:", "if player is whitelisted"})
-@Since("2.5.2")
+@Description("Whether or not the server or a player is whitelisted, or the server is whitelist enforced.")
+@Examples({
+	"if player is whitelisted:",
+	"if server is whitelisted:",
+	"if server is whitelist enforced:"
+})
+@Since("2.5.2, INSERT VERSION (enforce, offline players)")
+@RequiredPlugins("MC 1.17+ (enforce)")
 public class CondIsWhitelisted extends Condition {
-	
+
+	private static final boolean ENFORCE_SUPPORT = Skript.methodExists(Bukkit.class, "isWhitelistEnforced");
+
 	static {
-		Skript.registerCondition(CondIsWhitelisted.class,
-			"[the] server (is|1¦is(n't| not)) white[ ]listed",
-			"%players% (is|are)(|1¦(n't| not)) white[ ]listed");
+		String[] patterns = new String[ENFORCE_SUPPORT ? 3 : 2];
+		patterns[0] = "[the] server (is|not:(isn't|is not)) white[ ]listed";
```

Comment:
Should be `[in] white[ ]list[ed]`

Reply:
Wait, I'm confused. How does "if the server is in whitelist" or "if the server is in whitelisted" make any sense? Maybe "if the server is in whitelist mode", but without "mode" it doesn't make sense to me.

"if the server is whitelisted" seems fine.

Reply:
Due to Lime's addition it's incorrect to use `is in whitelisted` but giving the option to `[ed]` will allow it to be `is in whitelist` however it will not be 100% perfect as it allows what you mentioned `is whitelist` which is wrong

Adding `[mode]` will make it more correct or even better
`[the] server (is|not:(isn't|is not)) (in white[ ]list mode|white[ ]listed)`

Reply:
Ok that makes more sense. Just making sure that `server is in whitelist` doesn't really work because the server can't be IN the whitelist, it's players that are in the whitelist. The server can only be whitelisted or be in whitelist mode.